src/validators/dataclass.rs

```diff
@@ -595,7 +624,14 @@ impl DataclassValidator {
         input: &'data impl Input<'data>,
     ) -> ValResult<'data, ()> {
         let (dc_dict, post_init_kwargs): (&PyAny, &PyAny) = val_output.extract(py)?;
-        force_setattr(py, dc, intern!(py, "__dict__"), dc_dict)?;
+        if self.slots.is_some() {
```

Comment:
Similar issue here where you probably want to know which fields correspond to slots and which don't and handle appropriately.

Reply:
I think there's a reasonable world where we tell people "if you use slots in your pydantic dataclasses, then all subclasses must be slots dataclasses, and all fields must have slots". But if we do that, then I think we need to tell people up front that their type has a problem, rather than delaying the issue to runtime validation, or worse, silently not fully initializing the objects.

But I think it would probably be preferable if we just handled mixed slots/`__dict__` dataclasses properly automatically and didn't force handling on users